src/pmpo_MPMesh_assembly.hpp

```diff
@@ -0,0 +1,325 @@
+#ifndef POLYMPO_ASSEMBLY_H
+#define POLYMPO_ASSEMBLY_H
+
+#include "pmpo_wachspressBasis.hpp"
+
+namespace polyMPO{
+
+DoubleView MPMesh::assemblyV0(){
+    int numVtxs = p_mesh->getNumVertices();
+    auto elm2VtxConn = p_mesh->getElm2VtxConn();
+    
+    DoubleView vField("vField2",numVtxs);
+    auto mpPositions = p_MPs->getData<MPF_Cur_Pos_XYZ>(); //get the array of MP coordinates/positions
+    auto assemble = PS_LAMBDA(const int& elm, const int& mp, const int& mask) {
+    //for elm in elementsInMesh { //pseudo code - the 'parallel_for' handles this
+    //  for mp in materialPointsInElm { //pseudo code (cont.)
+          if(mask) { //if material point is 'active'/'enabled'
+            int nVtxE = elm2VtxConn(elm,0); //number of vertices bounding the element
+            for(int i=0; i<nVtxE; i++){
+              int vID = elm2VtxConn(elm,i+1)-1; //vID = vertex id
+              double distance = mpPositions(mp,0) + mpPositions(mp,1) + mpPositions(mp,2);
+              Kokkos::atomic_add(&vField(vID),distance);
+            }
+          }
+    //  }
+    //}
+    };
+    p_MPs->parallel_for(assemble, "assembly");
+    return vField;
+}
+
+template <MeshFieldIndex meshFieldIndex>
+void MPMesh::assemblyVtx0(){
+  Kokkos::Timer timer;
+  constexpr MaterialPointSlice mpfIndex = meshFieldIndexToMPSlice<meshFieldIndex>;
+  auto elm2VtxConn = p_mesh->getElm2VtxConn();  
+  auto mpData = p_MPs->getData<mpfIndex>();
+  const int numEntries = mpSliceToNumEntries<mpfIndex>();
+
+  int numVtx = p_mesh->getNumVertices();
+  p_mesh->fillMeshField<meshFieldIndex>(numVtx, numEntries, 0.0);
+  auto meshField = p_mesh->getMeshField<meshFieldIndex>();
+  auto weight = p_MPs->getData<MPF_Basis_Vals>();
+
+  const double zero = 0.0;
+  Kokkos::View<double*> sumWeights("sumWeights", numVtx);
+  auto assemble = PS_LAMBDA(const int& elm, const int& mp, const int& mask) {
+    if(mask) { //if material point is 'active'/'enabled'
+      int nVtxE = elm2VtxConn(elm,0); //number of vertices bounding the element
+      for(int i=0; i<nVtxE; i++){
+        int vID = elm2VtxConn(elm,i+1)-1; //vID = vertex id
+        double fieldComponentVal;
+        Kokkos::atomic_add(&sumWeights(vID), weight(mp, 0));
+        for(int j=0;j<numEntries;j++){
+          fieldComponentVal = mpData(mp,j) * weight(mp, 0);
+          Kokkos::atomic_add(&meshField(vID,j),fieldComponentVal);
+        }
+      }
+    }
+  };
+  p_MPs->parallel_for(assemble, "assembly");
+  Kokkos::MDRangePolicy<Kokkos::Rank<2>> policy({0,0},{numVtx, numEntries});
+  Kokkos::parallel_for("assembly average", policy, KOKKOS_LAMBDA(const int vtx, const int entry){
+    if (sumWeights(vtx) != zero) 
```

Comment:
This needs to be a tolerance, but setting an appropriate value of tolerance will require some work. So lets open an issue on this for now.